### ISSUE TITLE
Rake instrumentation relies on SDK method instead of API only

### ISSUE THREAD

**Issue by richardmcmillen** [open-telemetry/opentelemetry-ruby-contrib#190]
Similarly to the [ActiveJob issue](https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/189) and [this discussion:](https://github.com/open-telemetry/opentelemetry-ruby/discussions/1401#discussioncomment-4128761) 

The ActiveJob instrumentation relies on an SDK method which is not part of the API: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/rake/lib/opentelemetry/instrumentation/rake/patches/task.rb#L32

The SDK implementation: https://open-telemetry.github.io/opentelemetry-ruby/opentelemetry-sdk/v1.2.0/OpenTelemetry/SDK/Trace/TracerProvider.html#force_flush-instance_method

The API implementation:

- https://open-telemetry.github.io/opentelemetry-ruby/opentelemetry-api/v1.1.0/OpenTelemetry/Trace/TracerProvider.html
- https://open-telemetry.github.io/opentelemetry-ruby/opentelemetry-api/v1.1.0/OpenTelemetry/Internal/ProxyTracerProvider.html

**Impact**

Not much, as this code path is only ever hit when the instrumentation is installed from the registry, and this would typically only happen when using the SDK.

However, this is confusing as the gem depends on the API and not the SDK according to the gemspec: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/rake/opentelemetry-instrumentation-rake.gemspec

**Comment by ahayworth** [comment]
Waiting on the outcome of https://github.com/open-telemetry/opentelemetry-ruby/discussions/1401 and https://github.com/open-telemetry/opentelemetry-specification/issues/2944 before taking action, but once a direction is established I can clean this up.
